applications: nrf_desktop: Document using MCUboot+XIP in configurations (#11182)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

* Updated:

  * The nRF desktop configurations that enable :ref:`nrf_desktop_bluetooth_guide_fast_pair`.
    The configurations use the MCUboot bootloader built in the direct-xip mode (``MCUBOOT+XIP``) instead of the ``B0`` bootloader.
    This is done to support firmware updates using both :ref:`nrf_desktop_dfu` and :ref:`nrf_desktop_dfu_mcumgr` modules.

>   * The :ref:`nrf_desktop_dfu_mcumgr` is used instead of :ref:`nrf_desktop_smp` in MCUboot SMP configuration (:file:`prj_mcuboot_smp.conf`) for the nRF52840 DK.

Reviewer: ```suggestion
  * The :ref:`nrf_desktop_dfu_mcumgr` is used instead of the :ref:`nrf_desktop_smp` in MCUboot SMP configuration (:file:`prj_mcuboot_smp.conf`) for the nRF52840 DK.
```

* Updated:

>   * The nRF desktop configurations that enable :ref:`nrf_desktop_bluetooth_guide_fast_pair` use MCUboot bootloader built in the direct-xip mode (``MCUBOOT+XIP``) instead of ``B0`` bootloader.
>     This is done to support firmware updates using both :ref:`nrf_desktop_dfu` and :ref:`nrf_desktop_dfu_mcumgr`.

Reviewer 2: ```suggestion
  * The nRF desktop configurations that enable :ref:`nrf_desktop_bluetooth_guide_fast_pair`.
     The configurations use the MCUboot bootloader built in the direct-xip mode (``MCUBOOT+XIP``) instead of the ``B0`` bootloader.
    This is done to support firmware updates using both :ref:`nrf_desktop_dfu` and :ref:`nrf_desktop_dfu_mcumgr` modules.
```

>   * The :ref:`nrf_desktop_dfu_mcumgr` is used instead of :ref:`nrf_desktop_smp` in MCUboot SMP configuration (:file:`prj_mcuboot_smp.conf`) for the nRF52840 DK target.

Reviewer 2: ```suggestion
  * The :ref:`nrf_desktop_dfu_mcumgr` is used instead of :ref:`nrf_desktop_smp` in MCUboot SMP configuration (:file:`prj_mcuboot_smp.conf`) for the nRF52840 DK build targets.
```
nRF52840 DK build targets Or nRF52840 DK?

Author: `nRF52840 DK`
